app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_post_counter
```

Comment:
### Optional
- Kindly consider using the rails method `after_save` to automatically invoke the post counter method every time the model is being created or gets updated! 🙏 

> Keep this in mind in the other models as well 😄